config/auth_metadata_service.go

```diff
@@ -0,0 +1,169 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+)
+
+// timeout to wait for metadata service to respond
+const metadataServiceTimeout = 10 * time.Second
+
+const MetadataServiceVersion = "1"
+const MetadataServiceVersionHeader = "X-Databricks-Metadata-Version"
+const MetadataServiceHostHeader = "X-Databricks-Host"
+
+// response expected from the metadata service
+type serverResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+
+	// epoch in seconds since 1970-01-01T00:00:00Z
+	ExpiresOn json.Number `json:"expires_on"`
+}
+
+// Credentials provider that fetches a token from a locally running HTTP server
+//
+// The credentials provider will perform a GET request to the configured URL.
+//
+// The MUST return 4xx response if the "X-Databricks-Metadata-Version" header
+// is not set or set to a version that the server doesn't support.
+//
+// The server MUST guarantee stable sessions per URL path. That is, if the
+// server returns a token for a Host on a given URL path, it MUST continue to return
+// tokens for the same Host.
+//
+// The server MUST return a 4xx response if the Host passed in the "X-Databricks-Host"
+// header doesn't match the token.
+//
+// The server is expected to return a JSON response with the following fields:
+//
+// - access_token: The requested access token.
+// - token_type: The type of token, which is a "Bearer" access token.
+// - expires_on: Unix timestamp when the access token expires.
+
```

Comment:
remove the empty line between the type and the comment, if you want this comment to appear as documentation on pkg.go.dev